Modules/Intent.Modules.CosmosDB/Templates/CosmosDBDocument/CosmosDBDocumentTemplatePartial.cs

```diff
@@ -183,6 +184,24 @@ private void AddPropertiesForAggregate(CSharpClass @class)
                 }
             }
 
+            var useOptimisticConcurrency = ExecutionContext.Settings.GetCosmosDb().UseOptimisticConcurrencyDefault();
+            if (useOptimisticConcurrency)
+            {
+                // Etag implementation:
+                @class.AddField("string?", "etag", field =>
+                {
+                    field.Private();
+                    field.AddAttribute($"{UseType("Newtonsoft.Json.JsonProperty")}(\"_etag\")");
```

Comment:
I would like to revisit this in the future. If I add the attribute to the public property, it doesn't deserialize the etag correctly when reading from the repository. This is true if I do not add the public property either. The public can also not implement the `ICosmodDBDocument` interface, even though that interface implements `IItemWithEtag`. There is also a problem when adding the interface implementation for `Etag` on the `ICosmosDBDocument` interface.

I am not happy with this implementation, but it seems to be the only combination that works. The public property for `Etag` below is superfluous, and is never used. It SHOULD not break the functionality to retrieve the etag if I remove this.